src/Config.ts

```diff
@@ -102,19 +74,126 @@ export class Chain {
+   * @param config Facilitator config object.
+   */
+  private constructor(config: any) {
+    this.database = config.database || new DBConfig();
+    this.chains = config.chains || {};
+    this.encryptedAccounts = config.encryptedAccounts || {};
+  }
+
+  /**
+   * It writes facilitator config object.
+   * @param {string} chain Auxiliary chain id.
+   */
+  public writeToFacilitatorConfig(chain: string): void {
+    const mosaicConfigDir = Directory.getMosaicDirectoryPath();
+    const configPath = path.join(
+      mosaicConfigDir,
+      chain,
+    );
+    fs.ensureDirSync(configPath);
+
+    fs.writeFileSync(
+      path.join(configPath, FacilitatorConfig.fileName),
+      JSON.stringify(this, null, '    '),
+    );
+  }
+
+  /**
+   * This reads mosaic config from the json file and creates FacilitatorConfig object.
+   * @param {string} chain Auxiliary chain id.
+   * @returns {FacilitatorConfig} Facilitator config object.
    */
-  public constructor(readonly chainId: string, facilitatorConfig: Record<string, any>) {
-    const chainData = facilitatorConfig.chains[chainId];
-    this.rpc = chainData.rpc;
-    this.worker = new Account(
-      chainData.worker,
-      facilitatorConfig.encrypted_accounts[chainData.worker],
+  public static from(chain: string): FacilitatorConfig {
+    const facilitatorConfigPath = path.join(
+      Directory.getMosaicDirectoryPath(),
+      chain,
+      FacilitatorConfig.fileName,
     );
+
+    if (fs.existsSync(facilitatorConfigPath)) {
+      const config = fs.readFileSync(facilitatorConfigPath).toString();
+      if (config && config.length > 0) {
+        const jsonObject = JSON.parse(config);
+        return new FacilitatorConfig(jsonObject);
+      }
+    }
+    return new FacilitatorConfig({});
+  }
+
+  /**
+   * @returns {FacilitatorConfig} FacilitatorConfig object.
+   */
+  public static new(): FacilitatorConfig {
+    return new FacilitatorConfig({});
+  }
+
+  /**
+   * It verifies if facilitator config is present for a auxiliary chainid. If already present then it exits.
+   * @param {string} chain Auxiliary chain id.
+   */
+  public static assertNotExists(chain: string): void {
+    try {
+      const statOutput = fs.statSync(
+        path.join(Directory.getMosaicDirectoryPath(), chain, this.fileName),
+      );
+
+      if (statOutput.size > 0) {
+        Logger.error('facilitator config already present. use -f option to override the existing facilitator config.');
+        process.exit(1);
+      }
+    } catch (e) {
+      Logger.info('creating the facilitator config');
+    }
+  }
+
+  /**
+   * It returns origin chainid by searching through the mosaic config.
+   * @param {number} auxChainId Chain id of the auxiliary chain.
+   * @param {string} mosaicConfigPath Location where the mosaic config is present.
+   * @returns {number} returns chainid of the origin chain.
+   */
+  public static getOriginChainId(auxChainId: number, mosaicConfigPath: string): number {
```

Comment:
Why do we need this function?
mosaicConfig will have origin chain.
In facilitator init, just check if the aux chain exists or not.

Reply:
Moved the code to `facilitator-init` executable.